src/resource.js

```diff
@@ -0,0 +1,35 @@
+import { each, fromCallback } from 'bluebird'
+import { readFileAsync } from './util'
+import { Buffer } from 'buffer'
+import globs from 'globby'
+import { stat } from 'fs'
+
+async function isDirectoryAsync (path) {
+  const stats = await fromCallback(cb => stat(path, cb))
+  return stats.isDirectory()
+}
+
+export default async function resource (compiler, next) {
+  const resources = compiler.resources = {
+    index: {},
+    bundle: ''
+  }
+
+  if (!compiler.options.resources.length) {
+    return next()
+  }
+
+  await each(globs(compiler.options.resources), async (file) => {
+    if (await isDirectoryAsync(file)) {
+      return
+    }
+    const contents = await readFileAsync(file)
+    const encodedContents = contents.toString('base64')
```

Comment:
Could these files be stored as raw bytes instead of base64 encoding?  Seems like it would be a lot better size-wise.